storage/fs.go

```diff
@@ -85,27 +86,36 @@ func newFileSystemManager(
 		opts:     opts,
 		jitter:   jitter,
 		status:   fileOpNotStarted,
+		enabled:  true,
 	}, nil
 }
 
-func (m *fileSystemManager) ShouldRun(t time.Time) bool {
-	// If we haven't bootstrapped yet, no actions necessary.
-	if !m.database.IsBootstrapped() {
-		return false
-	}
+func (m *fileSystemManager) Disable() bool {
```

Comment:
Might it be worth returning just the status here instead of a boolean?  The status as it's an enum-ish seems more descriptive.  You could also add it to the return type of `Enable` too.